src/include/daos_types.h

```diff
@@ -257,6 +258,51 @@ daos_is_valid_uuid_string(const char *uuid)
 
 	return true;
 }
+
+static inline bool
+daos_is_valid_uuid_lower_string(const char *uuid)
+{
+	const char *p;
+	int         len = DAOS_UUID_STR_SIZE - 1; /* Not include the terminated '\0' */
+	int         i;
+
+	if (strnlen(uuid, len) != len)
+		return false;
+
+	for (i = 0, p = uuid; i < len; i++, p++) {
+		if (i == 8 || i == 13 || i == 18 || i == 23) {
+			if (*p != '-')
+				return false;
+		} else if (!isxdigit(*p) || isupper(*p)) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static inline bool
+daos_is_valid_uuid_upper_string(const char *uuid)
+{
+	const char *p;
+	int         len = DAOS_UUID_STR_SIZE - 1; /* Not include the terminated '\0' */
+	int         i;
+
+	if (strnlen(uuid, len) != len)
+		return false;
+
+	for (i = 0, p = uuid; i < len; i++, p++) {
+		if (i == 8 || i == 13 || i == 18 || i == 23) {
+			if (*p != '-')
+				return false;
```

Comment:
Since these two functions are basically identical, could they share most of their logic? The isupper/islower function could be passed as a function pointer.